### PR TITLE
Prepare Nimbus SDK for messaging transition

### PR DESCRIPTION
**Pull request by jhugman**
Fixes [EXP-4284](https://mozilla-hub.atlassian.net/browse/EXP-4284).

This PR fixes a number of very small issues:

- renames GleanPlumb to NimbusMessaging or NimbusHelper
- It implements a cache in the JEXL helper.
- it exposes event store to the apps via the NimbusMessaging protocol/interface.
- it adds a destroy() method to JEXL helper in Android.

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Review by brennie — APPROVED** [review]
Not a Kotlin or Swift person, but nothing seems out of place

File: components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusHelpers.kt
Comment (jeddai) [inline components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusHelpers.kt:73]: Are there tests covering these methods?

**Review by jeddai — APPROVED** [review]
r+wc 

There's a few methods that have been added to the interfaces — I'm approving so this doesn't need another review cycle, but could you please add tests for those or ensure they're covered by something before merging?

Otherwise, this change looks great!

File: components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusHelpers.kt
Comment (jhugman) [reply components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusHelpers.kt:73]: The `evalJexl` method is already tested in the [`NimbusMessagingTest`](https://github.com/mozilla/application-services/pull/6080/files#diff-e06c0ae9c4a583c0128b90d539310b186bdb62fe6295ba027d35a3dbe725d897).

Knowing this, @jeddai is happy to disregard.

**Pull request merged**
